tools/hooks/golang-precommit.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+set -e
+
+# Get the root directory of the project
+ROOT_DIR=$(git rev-parse --show-toplevel)
+
+# Change to the CLI directory where the Makefile is located
+cd "$ROOT_DIR/tools/cli"
+
+# Format Go files
+echo "Formatting Go files..."
+make fmt
+
+# Run golangci-lint
+echo "Running golangci-lint..."
+make lint
+
+# Run unit tests for Go files
+echo "Running Go unit tests..."
+make unit-test
```

Comment:
Since we have most of this in the cli makefile, can we have here 
```
#!/usr/bin/env bash
pushd tools/cli
make pre-commit
```

so that we reuse the logic in the make file in the cli